### DISCUSSION TITLE
Tempo 2.0 search perf

### DISCUSSION THREAD

**Discussion by kostya9** [grafana/tempo#2112] [Q&A] [answered]
Hi!

After upgrading Tempo from 1.5 to 2.0, I noticed that the data for this dropdown is loading much longer. When previously it felt almost instantaneous (seconds), now it loads in 30s+ (sometimes a couple of minutes). I am unsure whether this is connected to the new storage format or something else. Also, I noticed that searching by trace id is now taking longer.

The number of spans we store is not small, something like 3k spans a second.

![image](https://user-images.githubusercontent.com/3115913/219692789-a468461b-202e-4ddc-97a0-e8a37b9e21e3.png)

Do you have any advice on how to investigate the root cause? What metrics to look at, what configuration parameters to tweak?

UPD: In logs we see "error querying ingesters in Querier.SearchTags: failed to get response from ingesters: context deadline exceeded"

**Comment by kostya9** [comment, marked as answer]
We tinkered with config a little, and the problem disappeared.

FYI The candidate for the culprit is the `reportingEnabled` setting that was accidentally enabled with the upgrade, but we are not sure at all.

**Reply by joe-elliott** [reply]
it seems very odd that `reportingEnabled` impacted your searches. if it did that's concerning and we'd like to fix it. can you confirm by toggling that setting back on/off that it did, in fact, impact search?

**Reply by kostya9** [reply]
Today I saw 10s+ times for this dropdown population too.

TBH, upon thinking about it, I am thinking that this (tag search time improvement with minor config changes) was coincidental/transient, and the root problem is the same as in https://github.com/grafana/tempo/discussions/2118 - just search for tag values being slow-ish (as far as I understand, this dropdown just issues a search for service name tag values).

**Reply by joe-elliott** [reply]
yup, we are definitely seeing tag search times take too long in certain circumstances. we are working to improve.
